Sources/PointFree/GitHub.swift

```diff
@@ -3,27 +3,39 @@ import Foundation
 import Optics
 import Prelude
 
-public struct GitHubAccessToken: Codable {
-  let accessToken: String
+public struct GitHub {
+  /// Fetches an access token from GitHub from a `code` that was obtained from the callback redirect.
+  var fetchAuthToken: (String) -> EitherIO<Prelude.Unit, AccessToken>
 
-  enum CodingKeys: String, CodingKey {
-    case accessToken = "access_token"
+  /// Fetches a GitHub user from an access token.
+  var fetchUser: (AccessToken) -> EitherIO<Prelude.Unit, User>
+
+  static let `default` = GitHub(
+    fetchAuthToken: PointFree.fetchAuthToken,
+    fetchUser: PointFree.fetchUser
+  )
+
+  public struct AccessToken: Codable {
+    let accessToken: String
+
+    enum CodingKeys: String, CodingKey {
+      case accessToken = "access_token"
+    }
   }
-}
 
-public struct GitHubUser: Codable {
-  let email: String
-  let id: Int
-  let name: String
-}
+  public struct User: Codable {
```

Comment:
Nice to be able to nest these types and have both `Database.User` and `GitHub.User`.

Reply:
yes!